src/peakrdl_python/templates/peakrdl_python/base.py

```diff
@@ -36,8 +39,28 @@ def inst_name(self) -> str:
         """
         return self.__inst_name
 
+    @property
+    def parent(self) -> Optional[Base]:
+        """
+        parent of the node or field, or None if it has no parent
+        """
+        return self.__parent
+
+    @property
+    def full_name(self) -> str:
+        """
+        The full hierarchical name of the instance
+        """
+        return_name = self.inst_name
+        p = self.parent
+        while p is not None :
+            return_name = p.inst_name + "." + return_name
```

Comment:
@mpriestleyidex 

If this function has to provide the "full" name surely this would be p.full_name + '.' self.inst_name, otherwise with will only go one layer back

Reply:
Good point....

Reply:
I have made this change